### PR TITLE
[CI/942]: Update actions checkout version from v2 to v3

### LINKED ISSUE #942: Outdated checkout step version
**What were you trying to achieve?**
Run a GitHub action step that involves checking out the codebase

**What are the expected results?**
The checkout to run without any errors or warnings.

**What are the received results?**
The checkout step run with the warning as below
image

**What are the steps to reproduce the issue?**
Checkout [this workflow run](https://github.com/airqo-platform/AirQo-frontend/actions/runs/3443565374) and scroll to the bottom

**Additional context**
- This issue is cause by use of an outdated version(v2 instead of v3) of the [actions/checkout](https://github.com/actions/checkout) GitHub app so the solution would simply involve updating this version.
- The ideal solution updates this version in all [these workflow files](https://github.com/airqo-platform/AirQo-frontend/tree/staging/.github/workflows)
- A sample line to update is below
  https://github.com/airqo-platform/AirQo-frontend/blob/b084cca08fdfeffd2f10b8e93b2d008204c8417c/.github/workflows/deploy-k8s-changes-to-staging.yml#L23

### PR DESCRIPTION
**Pull request by Vinitvh**
#### Summary of Changes (What does this PR do?)

- Updated actions checkout version from `v2` to use `v3` following `Nodejs 12 actions are deprecated` warning from GitHub.
- Updated `setup-node` to `v3` with node version `16`

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [x] I've included the issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Static code analysis

#### What are the relevant tickets?

- Closes #942

### CODE REVIEW

File: .github/workflows/deploy-frontend-pr-previews.yml [hunk 1]
```diff
@@ -97,9 +97,9 @@ jobs:
         run: gcloud secrets versions access latest --secret="sta-env-netmanager" > netmanager/.env
 
       - name: npm Install and Build
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
-          node-version: "12"
+          node-version: "16"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontend-pr-previews.yml:102]: This one

File: .github/workflows/deploy-frontends-to-production.yml [hunk 2]
```diff
@@ -63,9 +63,9 @@ jobs:
         run: gcloud secrets versions access latest --secret="prod-env-netmanager" > netmanager/.env
 
       - name: NPM Setup and Build
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
-          node-version: "12"
+          node-version: "16"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontends-to-production.yml:68]: This one too

File: .github/workflows/deploy-frontends-to-staging.yml [hunk 3]
```diff
@@ -255,9 +255,9 @@ jobs:
         run: gcloud secrets versions access latest --secret="sta-env-calibrate-app" > calibrate/.env
 
       - name: NPM Setup and Build
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
-          node-version: "12"
+          node-version: "16"
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-frontends-to-staging.yml:260]: Here too

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
Thanks @Vinitvh 

I however have a small request. Could we keep the node version we are using to build the docker images 12 given that the main complaint only came from the checkout action.

Thus, kindly revert the changes to the (actions/setup-node@v2)sections below.

image

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @Vinitvh 
LGTM

**Review by Baalmart — APPROVED** [review]
LGTM, thanks @Vinitvh

**Pull request merged**